chore: Mark Gusto 'gender' field as deprecated (#239)
Finch-API/finch-docs: docs/Development-Guides/Field-Support/Compatibility.md

<p><i>¹ Initial only</i><br>
<i>² No work email available</i><br>
<i>³ Contractors don't have these fields in the QBO system</i></p>

> <i>⁴ This field has been deprecated and should not be relied upon for accuracy</i></p>

Author: Any other suggestions on phrasing?

> <i>⁴ This field has been deprecated and should not be relied upon for accuracy</i></p>

Author: ```suggestion
<i>⁴ This field is deprecated and may be inaccurate</i></p>
```
